os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_central/ble_tizenrt_client.c

```diff
@@ -74,6 +77,167 @@ trble_result_e rtw_ble_client_init(trble_client_init_config* init_parm)
+        whitelist_arg->remote_bd_type = GAP_REMOTE_ADDR_LE_RANDOM;    /* Random Address */
+    } else {
+        os_mem_free(whitelist_arg);
+        debug_print("\n[%s] Unknown Address Type", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+
+    if (ble_tizenrt_client_send_msg(BLE_TIZENRT_MODIFY_WHITELIST, whitelist_arg) == false)
+    {
+        os_mem_free(whitelist_arg);
+        debug_print("\r\n[%s] msg send fail", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    if (false == os_mutex_take(ble_tizenrt_modi_whitelist_sem,3000))
+    {
+        debug_print("\r\n[%s] set whitelist timeout", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    if (modify_whitelist_code != 0)
+    {
+        debug_print("\r\n[%s] set whitelist fail", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    return TRBLE_SUCCESS;
+}
+
+trble_result_e rtw_ble_client_scan_whiltelist_delete(trble_addr *addr)
+{
+    if (client_init_parm == NULL || client_init_parm->trble_device_scanned_cb == NULL || addr == NULL)
+    {
+        return TRBLE_FAIL;
+    }
+
+    if (ble_tizenrt_modi_whitelist_sem == NULL)
+    {
+        if(false == os_mutex_create(&ble_tizenrt_modi_whitelist_sem))
+        {
+            printf("\r\n[%s] create sem fail!", __FUNCTION__);
+            return TRBLE_FAIL;
+        } else {
+            debug_print("\r\n[%s] create sem 0x%x success", __FUNCTION__, ble_tizenrt_modi_whitelist_sem);
+        }
+    }
+
+    T_TIZENRT_MODI_WHITELIST_PARAM *whitelist_arg = os_mem_alloc(0, sizeof(T_TIZENRT_MODI_WHITELIST_PARAM));
+    if (whitelist_arg == NULL)
+    {
+        debug_print("\n[%s] Memory allocation failed", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+
+    memcpy(whitelist_arg->remote_bd, addr->mac, GAP_BD_ADDR_LEN);
+    whitelist_arg->type = GAP_WHITE_LIST_OP_REMOVE;
+    if (addr->type == TRBLE_ADDR_TYPE_PUBLIC)
+    {
+        whitelist_arg->remote_bd_type = GAP_REMOTE_ADDR_LE_PUBLIC;    /* Public Address */
+    } else if ((TRBLE_ADDR_TYPE_RANDOM_STATIC <= addr->type) && (TRBLE_ADDR_TYPE_RANDOM_NON_RESOLVABLE >= addr->type)) {
+        whitelist_arg->remote_bd_type = GAP_REMOTE_ADDR_LE_RANDOM;    /* Random Address */
+    } else {
+        os_mem_free(whitelist_arg);
+        debug_print("\n[%s] Unknown Address Type", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+
+    if (ble_tizenrt_client_send_msg(BLE_TIZENRT_MODIFY_WHITELIST, whitelist_arg) == false)
+    {
+        os_mem_free(whitelist_arg);
+        debug_print("\r\n[%s] msg send fail", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    if (false == os_mutex_take(ble_tizenrt_modi_whitelist_sem,3000) == false)
+    {
+        debug_print("\r\n[%s] set whitelist timeout", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    if (modify_whitelist_code != 0)
+    {
+        debug_print("\r\n[%s] set whitelist fail", __FUNCTION__);
+        return TRBLE_FAIL;
+    }
+    return TRBLE_SUCCESS;
```

Comment:
If message send is success it will be free in ble_tizenrt_central_handle_upstream_msg()/ble_tizenrt_scatternet_handle_upstream_msg(), under case BLE_TIZENRT_MODIFY_WHITELIST.